### DISCUSSION TITLE
GitHub Sample Understanding & Build Pipeline

### DISCUSSION THREAD

**Discussion by Riaz-Dynamics365FOAndCommerce** [microsoft/Dynamics365Commerce.Solutions#206] [Q&A]
1. Is build pipeline only required to implement project (Store Commerce) to UAT/Production. On Local Dev-Machine (VM) while extending, debugging store commerce GitHub samples (CommerceScaleUnit, CommerceInStore) build pipeline did not need OR optional OR mandatory?

2. In commerce SDK why say POS extension (MPOS) as Independant Packaging Model. 
3. Can We use GitHub CommerceScaleUnit samples to build (POS, CRT & Retail Server (APIS) means whole store commerce project.
4. GitHub CommerceInStore sample contains multiple projects. which one is best to use for MPOS POS (using Real Time Service to connect with HQ (D365 Finance & Operation)) migration to commerce SDK. 
5. Share best guide to understand store commerce github sample code.

**Comment by madyke** [comment]
Hi @Riaz-Dynamics365FOAndCommerce - 

1. Correct, build pipelines are not needed for local testing with Store Commerce. You can download the latest version from LCS and use that to test your extensions. You do not need a build pipeline to bundle your extensions with Store Commerce to test them (more details below).

2. Commerce SDK uses an Independent Packaging Model, meaning that the extensions do not have to be bundled with Store Commerce for installation, the way they did with MPOS. Extensions can now be developed and installed seperately from the normal Store Commerce product. For more details, refer to the [MPOS to Store Commerce migration documentation](https://learn.microsoft.com/en-us/dynamics365/commerce/dev-itpro/pos-extension/migrate-mpos-store-commerce) 

3. @Pikhulya may be able to answer this since it's related to CSU samples, but I suspect he may need more info for what exactly you're trying to accomplish.

4. The InStore samples are meant to demonstrate specific extension functionality or scenarios, there isn't any one sample that provides a template for migration. The best place to start is the overall migration documentation I mentioned above and the documentation on [how to migration an extension from Retail SDK to Commerce SDK](https://learn.microsoft.com/en-us/dynamics365/commerce/dev-itpro/pos-extension/migrate-pos-extension).

5. Each sample in the InStore GitHub repo has a README that provides an overview of what the sample is, how to set it up, and what extensions/code it's demonstrating. For example, the [Triggers samples README](https://github.com/microsoft/Dynamics365Commerce.InStore/blob/release/9.47/src/StoreCommerceSamples/PosExtensionSamples/Pos/Extend/Triggers/readme.md) details how each trigger is used.
